src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -30,19 +30,17 @@ struct OperationParameters {
     var header: Params
     var query: Params
     var path: [KeyValueViewModel]
-    var hasOptionalsParams: Bool = false
+    var body: BodyParams?
+    var signature: [ParameterViewModel]
+
+    var hasOptionalParams: Bool
 
     /// Build a list of required and optional query params and headers from a list of parameters
-    init(
-        parameters: [Parameter]?,
-        operation: Operation,
-        operationParameters: OperationParameters? = nil
-    ) {
-        self.header = operationParameters?.header ?? Params()
-        self.query = operationParameters?.query ?? Params()
-        self.path = operationParameters?.path ?? [KeyValueViewModel]()
-
-        for param in parameters ?? [] {
+    init(parameters: [ParameterType], operation: Operation) {
```

Comment:
Why we don't need to pass in operationParameters anymore?

Reply:
`Operation.allParams` collects any parameters from the Request, which is why we used to have this. We would collect the params from the operation and then go back through and make a copy, updating with the params from the Request.